repos/base-hw/src/core/spec/x86_64/muen/platform_support.cc

```diff
@@ -100,11 +100,22 @@ bool Platform::get_msi_params(const addr_t mmconf, addr_t &address,
 
 Native_region * Platform::_ram_regions(unsigned const i)
 {
-	static Native_region _regions[] =
-	{
-		{ 25*1024*1024, 256*1024*1024 }
-	};
-	return i < sizeof(_regions)/sizeof(_regions[0]) ? &_regions[i] : 0;
+	if (i)
+		return 0;
+
+	static Native_region result = { .base = 0, .size = 0 };
+
+	if (!result.size) {
+		struct Sinfo::Memregion_info region;
+		if (!sinfo()->get_memregion_info("ram", &region)) {
+			PERR("Unable to retrieve base-hw ram region");
+			return 0;
+		}
+
+		result = { .base = region.address, .size = region.size };
+		PDBG("RAM base 0x%lx, size 0x%lx", result.base, result.size);
```

Comment:
In fact we can leave it out here and instead enable "verbose" flag in repos/base-hw/src/core/platform.cc, which prints out that information too. The "verbose" flag will be replaced soon, so that the information gets displayed everytime.